examples/vae/train.py

```diff
@@ -145,59 +70,41 @@ def eval_model(vae):
     metrics = compute_metrics(recon_images, images, mean, logvar)
     return metrics, comparison, generate_images
 
-  return nn.apply(eval_model, model())({'params': params})
-
-
-def prepare_image(x):
-  x = tf.cast(x['image'], tf.float32)
-  x = tf.reshape(x, (-1,))
-  return x
-
+  return nn.apply(eval_model, models.model(flags))({'params': params})
 
-def main(argv):
-  del argv
-
-  # Make sure tf does not allocate gpu memory.
-  tf.config.experimental.set_visible_devices([], 'GPU')
 
+def train_and_evaluate(flags):
```

Comment:
also here, I would propose to pass in the individual values instead, to make the interface better defined.